Tools/TestWebKitAPI/Tests/WebKitCocoa/ContentFiltering.mm

```diff
@@ -108,6 +109,52 @@ - (void)encodeWithCoder:(NSCoder *)coder
     return configuration;
 }
 
+@interface LoadAlternateNavigationDelegate : NSObject <WKNavigationDelegate>
+@end
+
+@implementation LoadAlternateNavigationDelegate
+
+- (void)webView:(WKWebView *)webView didFailProvisionalNavigation:(WKNavigation *)navigation withError:(NSError *)error
+{
+    EXPECT_WK_STREQ(WebKitErrorDomain, error.domain);
+    EXPECT_EQ(kWKErrorCodeFrameLoadBlockedByContentFilter, error.code);
+    [webView _loadAlternateHTMLString:@"FAIL" baseURL:nil forUnreachableURL:[error.userInfo objectForKey:NSURLErrorFailingURLErrorKey]];
+}
+
+- (void)webView:(WKWebView *)webView didFinishNavigation:(WKNavigation *)navigation
+{
+    [webView evaluateJavaScript:@"document.body.innerText" completionHandler:^(id result, NSError *error) {
+        EXPECT_TRUE([result isKindOfClass:[NSString class]]);
+        EXPECT_WK_STREQ(@"blocked", result);
+        isDone = true;
+    }];
+}
+
+- (void)webViewWebContentProcessDidTerminate:(WKWebView *)webView
+{
+    EXPECT_FALSE(true);
+}
+
+@end
+
+static void loadAlternateTest(Decision decision, DecisionPoint decisionPoint)
```

Comment:
Is it necessary to move this function?

Reply:
In order to make the test able to find all the functions where I initially placed it, yes. But I think I can just add the new test to the bottom to avoid this. I will make that change as not to pollute the blame too much.